pkg/build/api/validation/validation.go

```diff
@@ -311,7 +332,23 @@ func validateTrigger(trigger *buildapi.BuildTriggerPolicy) fielderrors.Validatio
 	case buildapi.ImageChangeBuildTriggerType:
 		if trigger.ImageChange == nil {
 			allErrs = append(allErrs, fielderrors.NewFieldRequired("imageChange"))
+			break
+		}
+		if trigger.ImageChange.From == nil {
+			break
+		}
+		if kind := trigger.ImageChange.From.Kind; kind != "ImageStreamTag" && kind != "" {
+			invalidKindErr := fielderrors.NewFieldInvalid(
+				"imageChange.from.kind",
+				kind,
+				"only an ImageStreamTag type of reference is allowed in an ImageChange trigger.")
+			allErrs = append(allErrs, invalidKindErr)
+			break
+		}
+		if trigger.ImageChange.From.Kind == "" {
+			trigger.ImageChange.From.Kind = "ImageStreamTag"
```

Comment:
i'd rather force people to make this explicit.  implied defaults end up being conversion challenges later.

also manipulating the object in validation is a pattern we're trying to rid ourselves of.  if you want to default it, you can default it in conversion.